src/vsr/replica.zig

```diff
@@ -7659,6 +7610,92 @@ pub fn ReplicaType(
             self.send_do_view_change();
         }
 
+        fn view_change_update_view_headers(self: *Self) void {
+            // Either:
+            // - Transition from normal status.
+            // - Recovering from normal status.
+            // - Retired primary that didn't finish repair.
+            assert(self.status == .normal or
+                (self.status == .recovering and self.log_view == self.view) or
+                (self.status == .view_change and self.log_view == self.view));
+
+            const primary_repairing =
+                self.status == .view_change and self.log_view == self.view;
```

Comment:
Don't really love this special casing. I tried to make the primary to update its view headers right after it receives dvc quorum, but that doesn't seem to work --- the repairing primary might still have gaps in headers, which prevents it from generating a valid SV.